streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredWindowStore.java

```diff
@@ -103,6 +103,19 @@ private Bytes keyBytes(final K key) {
         return Bytes.wrap(serdes.rawKey(key));
     }
 
+    @Override
+    public V fetch(final K key, final long timestamp) {
+        final long startNs = time.nanoseconds();
+        byte[] result;
+        try {
+            result = inner.fetch(keyBytes(key), timestamp);
+        } finally {
+            metrics.recordLatency(this.fetchTime, startNs, time.nanoseconds());
+        }
+
+        return serdes.valueFrom(result);
```

Comment:
should we move this up into the `try{..}catch{..}` so the deserialization cost is included in the metric?

Reply:
Good catch!!